File: go/picking_a_winner.go
package main

import (
	"bufio"
	"fmt"
	"math/rand"
	"os"
	"strings"
	"time"
)

func main() {
	rand.NewSource(time.Now().Unix())
	scanner := bufio.NewScanner(os.Stdin)

	players := []string{}

	for {
		fmt.Print("Enter a name: ")
		scanner.Scan()
		name := strings.TrimSpace(scanner.Text())

		if name == "" {
			break
		}

		players = append(players, name)
	}

	fmt.Printf("The winner is... %s\n", shuffle(players))
}

func shuffle(array []string) string {
	num := rand.Intn(len(array))

	return array[num]
}

File: python/picking_a_winner.py
import random

names: list[str] = []


def pick_winner(names: list[str]) -> str:
    num = random.randint(0, len(names) - 1)
    return names[num]


while True:
    name = input("Enter a name: ").strip()

    if name == "":
        break

    names.append(name)


winner = pick_winner(names)
print(f"The winner is... {winner}")
